### DISCUSSION TITLE
unsuspected behavior on execution_timeout for KubernetesPodOperator

### DISCUSSION THREAD

**Discussion by esposip** [apache/airflow#29050] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

I'm using the KubernetesPodOperator from airflow.providers.cncf.kubernetes.operators.kubernetes_pod. When I set the execution_timeout the task fails but the pod on EKS remains running.

### What you think should happen instead

The pod is terminated only if I manually force the task to end.

### Operating System

MWAA

### Versions of Apache Airflow Providers

2.2.2

### Deployment

MWAA

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
Can you please upgrede to latest version provieded by MWAA please? There were a complete rewrite of K8S interface since 2.2.2 becaus of similar problems you experienced. The only way to get it fixed is to ... upgrade (we never release bug fixes to old feature branches, so your only chance is to upgrade anyway.  there is almost no code in 2.2.2 that would remain in newer versions.

I will convert that one to a discussion - waiting for the user to report back what happens after upgrade.

**Reply by GHGHGHKO** [reply]
Hi, @potiuk
I’m experiencing the same issue. I’m using MWAA version 2.10.3

```python
from datetime import datetime, timedelta

    KubernetesPodOperator(
        execution_timeout=timedelta(minutes=5),  # todo doesn't work
...
)
```

Even after 5 minutes, pod on EKS remains running.
